test/Interop/Cxx/value-witness-table/basic.swift

```diff
@@ -0,0 +1,134 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-cxx-interop)
+//
+// REQUIRES: executable_test
+
+import Basic
+import StdlibUnittest
+
+var CXXDestructorTestSuite = TestSuite("CXXDestructor")
+
+protocol InitWithPtr {
+  init(value: UnsafeMutablePointer<Int32>!)
+}
+
+extension HasUserProvidedDestructor : InitWithPtr { }
+
+protocol InitWithBox {
+  init(box: HasUserProvidedDestructor)
+}
+
+extension EmptyDestructor : InitWithBox { }
+
+@inline(never)
```

Comment:
What's the significance of `@inline(never)`?

Reply:
If this is inlined into where it's used then `box`'s destructor may not be invoked until after the assertion in the caller.

Reply:
I'll add a comment.